packages/ledger/src/index.ts

```diff
@@ -1,279 +1,45 @@
-                  message: `The Provider does not support the requested method: ${method}`
-                })
-              }
-
-              // @ts-ignore
-              if (method === 'eth_sendTransaction') {
-                // @ts-ignore
-                return this.connector.sendTransaction(params[0])
-              }
-
-              // @ts-ignore
-              if (method === 'eth_signTransaction') {
-                // @ts-ignore
-                return this.connector.signTransaction(params[0])
-              }
-
-              // @ts-ignore
-              if (method === 'personal_sign') {
-                // @ts-ignore
-                return this.connector.signPersonalMessage(params)
-              }
-
-              // @ts-ignore
-              if (method === 'eth_sign') {
-                // @ts-ignore
-                return this.connector.signMessage(params)
-              }
-
-              // @ts-ignore
-              if (method.includes('eth_signTypedData')) {
-                // @ts-ignore
-                return this.connector.signTypedData(params)
-              }
-
-              if (method === 'eth_accounts') {
-                return this.connector.sendCustomRequest({
-                  id: 1337,
-                  jsonrpc: '2.0',
-                  method,
-                  params
-                })
-              }
-
-              const chainId = await this.request({ method: 'eth_chainId' })
-
-              if (!this.providers[chainId]) {
-                const currentChain = chains.find(({ id }) => id === chainId)
+export type LedgerOptionsWCv2 = {
+  walletConnectVersion: 2
+  enableDebugLogs?: boolean,
+  projectId: string
+  requiredChains?: string[] | number[]
+  requiredMethods?: string[]
+  optionalMethods?: string[]
+  requiredEvents?: string[]
+  optionalEvents?: string[]
+}
 
-                if (!currentChain) {
-                  throw new ProviderRpcError({
-                    code: ProviderRpcErrorCode.CHAIN_NOT_ADDED,
-                    message: `The Provider does not have a rpcUrl to make a request for the requested method: ${method}`
-                  })
-                }
+export type LedgerOptions = (
+  | LedgerOptionsWCv1
+  | LedgerOptionsWCv2
+)
 
-                this.providers[chainId] = new StaticJsonRpcProvider(
-                  currentChain.rpcUrl
-                )
-              }
+export const isHexString = (value: string | number) => {
+  if (typeof value !== 'string' || !value.match(/^0x[0-9A-Fa-f]*$/)) {
+    return false
+  }
 
-              return this.providers[chainId].send(
-                method,
-                // @ts-ignore
-                params
-              )
-            }
-          }
-        }
+  return true
+}
 
-        return {
-          provider: new EthProvider({ chains, connector })
-        }
-      }
-    }
-  }
+function ledger(options?: LedgerOptions): WalletInit {
+  const walletConnectVersion = options?.walletConnectVersion || 1
```

Comment:
```suggestion
  const walletConnectVersion = options?.walletConnectVersion || 2
```
Defaults to version 2 of wallet connect